opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -266,13 +266,18 @@ class OTELResourceDetector(ResourceDetector):
     def detect(self) -> "Resource":
         env_resources_items = os.environ.get(OTEL_RESOURCE_ATTRIBUTES)
         env_resource_map = {}
+
         if env_resources_items:
-            env_resource_map = {
-                key.strip(): value.strip()
-                for key, value in (
-                    item.split("=") for item in env_resources_items.split(",")
-                )
-            }
+            for item in env_resources_items.split(","):
+                if item.find("=") == -1:
+                    logger.warning(
```

Comment:
@NathanielRN I like the improved error message, whether it's an exception/warning/log :) The error handling principles allow the SDK to fail fast during initialization, so we could raise an exception or use the warnings mechanism @ocelotl has described. I'm fine with logging for now, but just something to keep in mind

Reply:
Hm did you mean then that a bad attribute should be cause to fail initializing the SDK? That makes sense too, I had assumed that we don't want to fail completely if just a bad attribute exists and that a warning log has a good enough chance of being seen if anyone is surprised at it missing...

I think an error would be less convenient because then you can't write things like I said below: `OTEL_RESOURCE_ATTRIBUTES="new_k=new_v,$OTEL_RESOURCE_ATTRIBUTES"` with `OTEL_RESOURCE_ATTRIBUTES=""` originally. Does that use case make sense? This came up in this discussion https://github.com/open-telemetry/opentelemetry-lambda/pull/164#discussion_r736095319 because OTel Java already handles this case (without an exception I assume)...

Reply:
I updated the code to re-use this code you pointed me to

https://github.com/open-telemetry/opentelemetry-python/blob/a1903c939917c7a6129476f12c9f50a587b479e5/opentelemetry-api/src/opentelemetry/util/re.py#L52-L53

because it makes sense that we want to support having `=` in the value. I also updated the tests.

> The issue is maybe we could update this code to re-use the baggage parsing code. If we rely on this undocumented behavior, it would be dangerous to do that.

Yeah my hope of allowing `OTEL_RESOURCE_ATTRIBUTES="new_k=new_v,$OTEL_RESOURCE_ATTRIBUTES"` could be dangerous if we release `1.x` which supports it and then suddenly remove it in a `1.(x+1)` release because it would basically be a breaking change... I think it is worth it to support this since it follow the very common `PATH=/new/path:$PATH` format and would be useful for scripts that initialize OTel Python 🙂 But I understand if we want to block on a spec decision first.

Reply:
Nah I don't want to block anything, marking resolved